### ISSUE TITLE
[basic.def.odr] p10 What does function definition scope mean in this current draft?

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#5406]
[basic.def.odr] p10 says
>  where *this is considered to be introduced within the innermost enclosing class or non-lambda **function definition scope**

And the comment in the example after this rule also refers to "function definition scope"

>  error: n is not odr-usable due to intervening function definition scope

In the subclause [[basic.scope]](https://eel.is/c++draft/basic.scope), we do not define "function definition scope", we just have function parameter scope. Does "function definition scope" intend to refer to function parameter scope instead?

**Comment by jensmaurer** [comment]
@opensdh , replacing this with "function parameter scope" sounds reasonable. What do you think?

**Comment by opensdh** [comment]
The example plainly thinks that there is some distinction (compare `A::f` to `g`), but C++20 didn't have any such scope defined either, so I'm pretty sure it's fine to make this change.

**Comment by frederick-vs-ja** [comment]
The term "function definition scope" is introduced via [P0588R1](https://wg21.link/p0588r1) and not defined or used elsewhere.

@zygoloid, what's the meaning of "function definition scope"?

**Comment by xmh0511** [comment]
@frederick-vs-ja  "function definition scope" had been removed in P1787. P1787 seems to be invented after P0588R1?

**Comment by frederick-vs-ja** [comment]
> "function definition scope" had been removed in P1787.

Not true. P1787 didn't touch "function definition scope" (you can check [P1787R6](https://wg21.link/p1787r6) and [EDIT4398](https://wg21.link/edit4398)).

The (only two) occurences of "function definition scope" in the draft haven't been changed since their introduction (contained in [N4713](https://wg21.link/n4713)).

**Comment by xmh0511** [comment]
Yes, you're right. I just remembered it wrong. Except that "function definition scope" appears in [basic.def.odr] p10, it doesn't appear in other subclauses anymore.  If I remember correctly, "function definition scope" hasn't been defined since c++17(or earlier version). From the literal meaning, it intends to refer to a function parameter scope, which contains any scope introduced in the function definition.
